### PR TITLE
chore: Bump e2e framework version to latest main

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <benye@amazon.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

Fixes https://github.com/thanos-io/thanos/issues/5700

### CODE REVIEW

**Review by squat — APPROVED** [review]
Wow good work here @yeya24

**Comment by yeya24** [comment]
There is a bug on the e2e framework. Let's hold this first.

File: go.mod
Comment (yeya24) [inline go.mod:254]: I use my temporary fork for now to bring https://github.com/thanos-io/objstore/pull/28. 
This could help unblock the build as it still uses the old version.

**Review by bwplotka — DISMISSED** [review]
Thanks! 

Can we avoid obj store fork?

**Comment by yeya24** [comment]
> Can we avoid obj store fork?

Yeah defintely. I am waiting for review of https://github.com/thanos-io/objstore/pull/28 and then I can bring the latest main in

**Comment by yeya24** [comment]
Finally fix the E2E tests. PTAL again @bwplotka @squat

**Review by squat — APPROVED** [review]
Thanks @yeya24 🎉🎉🎉

**Pull request merged**
